source/slang/slang-options.cpp

```diff
@@ -707,6 +708,26 @@ struct OptionsParser
 
                     rawEntryPoints.add(rawEntryPoint);
                 }
+                else if (argStr == "-lang")
+                {
+                    String name;
+                    SLANG_RETURN_ON_FAIL(tryReadCommandLineArgument(sink, arg, &argCursor, argEnd, name));
+
+                    SourceLanguage sourceLanguage = findSourceLanguageByName(name);
+
+                    if (sourceLanguage == SourceLanguage::Unknown)
+                    {
+                        sink->diagnose(SourceLoc(), Diagnostics::unknownSourceLanguage, name);
+                        return SLANG_FAIL;
+                    }
+                    else
+                    {
+                        while ((*argCursor)[0] != '-' && argCursor != argEnd)
+                        {
+                            SLANG_RETURN_ON_FAIL(addInputPath(*argCursor++, sourceLanguage));
+                        }
```

Comment:
Hmm... This means that the `-lang` option will apply to an source files that *immediately* follow it on the command line, but not to subsequent files after an intervening option. So if I write:

    slangc -lang slang -o foo.dxbc foo.hlsl

Then the `-lang` option there will be silently ignored.

I don't think this is a deal-breaker for getting this functionality checked in to unblock other work, but I would argue that we should clean this up in one of two ways:

1. We could make the `-lang` option set a `defaultSourceLanguage` variable local to the options parser, but not look for subsequent source files. Instead, the ordinary path for detecting source file arguments on the command line would pass down `defaultSourceLanguage` to `addInputPath` (or the latter could use the variable directly)

2. We could keep the current logic, but count the number of source files `-lang` applied to and diagnose an error if it was zero.

I don't have a very strong opinion about which of those options I'd prefer, but I'm kind of leaning toward (1).

Reply:
Right, so I had thought about this scenario a bit and I actually did this intentionally. The idea was that you could override a group of shader files with the `-lang` option and then also compile some others based on the classic method (i.e. based on file extension). So you could have a list of, let's say, 20 `.hlsl` files, with 10 of them using slang language features, and pass them all in at once but only use the slang translation unit for the ones you wanted. It's a bit of an esoteric example, I know, since I don't know how common the use case is of having the command line compile multiple files at once with the same entry point and shader stage